lib/streamlit/runtime/uploaded_file_manager.py

```diff
@@ -11,324 +11,118 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 import io
-import threading
-from typing import Dict, List, NamedTuple, Tuple
+from abc import abstractmethod
+from typing import List, NamedTuple, Sequence
 
-from blinker import Signal
+from typing_extensions import Protocol
 
 from streamlit import util
-from streamlit.logger import get_logger
-from streamlit.runtime.stats import CacheStat, CacheStatsProvider
-
-LOGGER = get_logger(__name__)
+from streamlit.proto.Common_pb2 import FileURLs as FileURLsProto
+from streamlit.runtime.stats import CacheStatsProvider
 
 
 class UploadedFileRec(NamedTuple):
     """Metadata and raw bytes for an uploaded file. Immutable."""
 
-    id: int
+    file_id: str
     name: str
     type: str
     data: bytes
 
 
+class UploadFileUrlInfo(NamedTuple):
+    """Information we provide for single file in get_upload_urls"""
+
+    file_id: str
+    upload_url: str
+    delete_url: str
+
+
+class DeletedFile(NamedTuple):
+    """Represents a deleted file in deserialized values for st.file_uploader and
+    st.camera_input
+    Return this from st.file_uploader and st.camera_input deserialize (so they can
```

Comment:
nit: should have a newline between the docstring summary and the rest of it

Reply:
done